core/arch/interp.c

```diff
@@ -7247,7 +7248,9 @@ decode_fragment(dcontext_t *dcontext, fragment_t *f, byte *buf, /*IN/OUT*/ uint
     bool coarse_elided_ubrs = false;
     dr_isa_mode_t old_mode;
     /* for decoding and get_ibl routines we need the dcontext mode set */
-    bool ok = dr_set_isa_mode(dcontext, FRAG_ISA_MODE(f->flags), &old_mode);
+    DEBUG_DECLARE(bool ok =)
```

Comment:
Maybe we should have a macro for such patterns where the returned boolean is supposed to be checked only in debug mode.

Reply:
Could be confusing though if you had:
```C
DEBUG_CHECK_RET_TRUE(dr_set_isa_mode(dcontext, FRAG_ISA_MODE(f->flags), &old_mode));
```
Which the reader will probably think is only executed at all in debug build.
Would have to name it ALWAYS_RUN_BUT_CHECK_RETVAL_IN_DEBUG() or sthg?

Reply:
Just going to leave the code as-is.